Defined 2 more profiles for OKE runs, added support to run OKE tests in parallel and sequence runs

marinakog opened the pull request:
Defined 2 more profiles oke-parallel, oke-sequencial in addition to oke-cert.
Added more tests to run in OKE.
https://build.weblogick8s.org:8443/job/wko-oke-nightly-seqential/769
https://build.weblogick8s.org:8443/job/wko-oke-nightly-parallel/84
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/10955

anpanigr reviewed (commented):
Why we need 3 profiles. Can the tests on sequential profile can not be run in parallel. We should create one profile wko-oke-cert which should include all the tests which can be run in OKE environment ( both in parallel and sequential fashion)

The seqential profile (kind-sequential)  in regular run are those tests which can not be run in parallel.

marinakog commented:
> 

if we want to verify all tests in OKE, we also need different profiles: I did same logic, moved tests which can't be run in parallel to oke-seq , added tests to oke-parallel and kept oke-cert for shorter version just to verify major fixes or certification.

anpanigr reviewed (commented):
Can you create tags oke-cert, oke-sequential and oke-parallel in class and define the profile in pom.xml based on these tags instead of explicit inclusion of the class

anpanigr reviewed (approved):
lgtm

sankarpn commented on integration-tests/src/test/java/oracle/weblogic/kubernetes/ItElasticLoggingFluentd.java:
```diff
@@ -375,11 +376,11 @@ private static void createDomainCrAndVerify(String adminSecretName,
     final String logHomeRootPath = "/scratch";
     // create the domain CR
-
+    String imagePoolPolicy = OKE_CLUSTER ? "Always" : "IfNotPresent";
```
```suggestion
    String imagePoolPolicy = OKE_CLUSTER ? V1Container.ImagePullPolicyEnum.ALWAYS : V1Container.ImagePullPolicyEnum.IFNOTPRESENT;
```

marinakog replied:
fluent spec takes string , not enum type

sankarpn commented on integration-tests/src/test/java/oracle/weblogic/kubernetes/ItFmwDomainInPVUsingWDT.java:
```diff
@@ -73,12 +73,11 @@ class ItFmwDomainInPVUsingWDT {
   private static LoggingFacade logger = null;
 
   private static final String domainUid = "fmw-domainonpv-wdt";
+  private static final String DOMAINHOMEPREFIX = "/shared/" + domainUid + "/domains/";
```
can we include the namespace as well in the path to make it unique?

marinakog replied:
done

sankarpn commented on integration-tests/src/test/java/oracle/weblogic/kubernetes/actions/impl/primitive/WebLogicImageTool.java:
```diff
@@ -85,6 +87,48 @@ public boolean updateImage() {
 
+  /**
+   * Inspect an image using the params using WIT inspect command.
+   * @return output if the command succeeds
```
```suggestion
   * @return inspection output string if the command succeeds otherwise null
```

sankarpn commented on integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonMiiTestUtils.java:
```diff
@@ -1433,7 +1434,7 @@ public static void patchDomainClusterWithAuxImageAndVerify(String domainUid,
         .append(auxiliaryImageName)
         .append("\", ")
-        .append("\"imagePullPolicy\": \"IfNotPresent\" ")
+        .append("\"imagePullPolicy\": \"" + imagePoolPolicy + "\" ")
```
```suggestion
        .append("\"imagePullPolicy\": \"" + imagePullPolicy + "\" ")
```

marinakog commented on integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/CommonMiiTestUtils.java:
```diff
@@ -1422,7 +1423,7 @@ public static void patchDomainClusterWithAuxImageAndVerify(String domainUid,
           managedServerPodName, domainNamespace)));
     }
-
+    String imagePoolPolicy = OKE_CLUSTER ? "Always" : "IfNotPresent";
```
it takes String , not enum

sankarpn reviewed (approved):
LGTM
